### ISSUE TITLE
"NAN prediction in example ***, forcing 0.000000"when using VW's MF!

### ISSUE THREAD

**Issue by wai7niu8** [JohnLangford/vowpal_wabbit#795]
Hi, all:
I meet the same problem when using VW's MF, but the solution above seems can't solve my problem.
There are the params:
cat vw_input | ./vw -b 18 -q ui --rank 10 --l2 3 --learning_rate 0.015 --decay_learning_rate 0.9 --power_t 0 --passes 20 -f vw.model  --cache_file vw.cache
and the result is:
creating quadratic features for pairs: ui
using l2 regularization = 3
final_regressor = vw.model
Num weight bits = 18
learning rate = 0.015
initial_t = 1
power_t = 0
decay_learning_rate = 0.9
using cache_file = vw.cache
ignoring text input in favor of cache input
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.667343 0.667343            1            1.0   1.0000   0.1831       23
0.628490 0.589637            2            2.0   1.0000   0.2321       23
0.573036 0.517581            4            4.0   1.0000   0.2994       23
0.840122 1.107208            8            8.0   1.0000   0.3520       23
0.632574 0.425027           16           16.0   1.0000   0.4074       23
3.970465 7.308356           32           32.0   5.0000   0.6162       23
6.304581 8.638697           64           64.0   1.0000   1.1481       23
4.797502 3.290423          128          128.0   2.0000   0.8625       23
2.697726 0.597951          256          256.0   1.0000   0.6555       23
2.137785 1.577844          512          512.0   4.0000   0.7334       23
56.857509 111.577233         1024         1024.0   1.0000   0.5840       23
41.128324 25.399138         2048         2048.0   1.0000   0.5939       23
25.930501 10.732679         4096         4096.0   1.0000   1.1791       23
19.470009 13.009517         8192         8192.0   1.0000   0.7646       23
15.094596 10.719182        16384        16384.0   1.0000   0.7499       23
16.379301 17.664006        32768        32768.0   1.0000   0.5364       23
14.600711 12.822122        65536        65536.0   2.0000   0.6233       23
16.287760 17.974809       131072       131072.0   1.0000   0.7090       23
20.793497 25.299233       262144       262144.0   1.0000   0.7385       23
16.575091 12.356686       524288       524288.0   1.0000   0.6054       23
14.781751 12.988411      1048576      1048576.0   2.0000   0.7033       23
15.769941 16.758130      2097152      2097152.0   1.0000   0.9231       23
14.939491 14.109041      4194304      4194304.0   1.0000   0.7153       23
14.897151 14.854812      8388608      8388608.0   1.0000   0.7733       23
14.918367 14.939582     16777216     16777216.0   1.0000   0.8988       23
13.257977 13.257977     33554432     33554432.0   3.0000   1.8546       23 h
...
...
NAN prediction in example 39410337, forcing 0.000000
NAN prediction in example 39410338, forcing 0.000000
NAN prediction in example 39410339, forcing 0.000000
NAN prediction in example 39410339, forcing 0.000000
NAN prediction in example 39410340, forcing 0.000000
...
...
And it will return a useless model with all NaN coefficients.

**Comment by JohnLangford** [comment]
This looks like it is very deep into the optimization process on a fairly large file so this is not easily reproduced.   Do you have a more reproducible example?

**Comment by wai7niu8** [comment]
I agree with u, john. 
I tried many times(keeping the same VW's params), and found when the num of examples exceeds 270000, the NaN produced. But i can't reproduce it on a more smaller data set.

**Comment by JohnLangford** [comment]
I'd suggest trying lrq instead.

-John

**Comment by JohnLangford** [comment]
Closing this for now---not reproducible.  Please reopen if something reproducible appears.

**Comment by danmelamed** [comment]
On what platform did this bug appear?  I might have found a cause, but only on Windows.

**Comment by wai7niu8** [comment]
I got this bug on Linux, but I can't replicate it on a smaller data set.
